skimage/measure/profile.py

```diff
@@ -109,3 +124,229 @@ def _line_profile_coordinates(src, dst, linewidth=1):
+        dim_x = - (d_pln + d_col) / d_row
+        length_vector = np.sqrt(2 + dim_x ** 2)
+        col_width = slice_width = (linewidth - 1) / (2 * length_vector)
+        row_width = (linewidth - 1) * (dim_x / length_vector) / 2
+    else:
+        # try finding the solution to ix + jy + kz = 0 for x = 1 and z = 1
+        dim_y = - (d_row + d_pln) / d_col
+        length_vector = np.sqrt(2 + dim_y ** 2)
+        row_width = slice_width = (linewidth - 1) / (2 * length_vector)
+        col_width = (linewidth - 1) * (dim_y / length_vector) / 2
+
+    # we add one above because we include the last point in the profile
+    # (in contrast to standard numpy indexing)
+    length = np.ceil(np.linalg.norm([d_pln, d_row, d_col]) + 1)
+
+    line_col = np.linspace(src_col, dst_col, length)
+    line_row = np.linspace(src_row, dst_row, length)
+    line_pln = np.linspace(src_pln, dst_pln, length)
+
+    # find divisor to get only first half of array and center point if odd
+    first_half_index = np.int(np.ceil(np.float(linewidth)/2))
+    if first_half_index < 1:
+        first_half_index = 1
+
+    perp_rows = np.array([np.linspace(row_i - row_width, row_i + row_width,
+                                      linewidth)[:first_half_index] for row_i in line_row])
+    perp_cols = np.array([np.linspace(col_i - col_width, col_i + col_width,
+                                      linewidth)[:first_half_index] for col_i in line_col])
+    perp_pln = np.array([np.linspace(slice_i - slice_width, slice_i + slice_width,
+                                       linewidth)[:first_half_index] for slice_i in line_pln])
+
+    perp_array = np.array([perp_pln, perp_rows, perp_cols])
+
+    # rotate all sample points around the direction axis if linewidth is > 1
+    if linewidth > 1:
+        perp_array = rotate_sample_points(linewidth, perp_array, src, dst)
+
+    return perp_array
+
+
+def rotate_sample_points(linewidth, perp_array, src, dst):
+    """Return the evenly rotated coordinates of the sample points along a scan line in 3d
+
+    Parameters
+    ----------
+    linewidth : int
+        Width of the scan, perpendicular to the line,
+        since the line is 3d, this value is the diameter
+        of a 3d cylinder following the scan line.
+    perp_array, shape (3, M, N), float
+        The coordinates of the profile along the scan line. The length of the
+        profile is the ceil of the computed length of the scan line.
+        The coordinates are 180 degrees apart.
+    src : 3-tuple of numeric scalar (float or int)
+        A first point where the line is passing through
+    dst : 3-tuple of numeric scalar (float or int)
+        A second point where the line is passing through
+
+    Returns
+    -------
+    sampling_array : array, shape (3, M, N), float
+        The coordinates of the 3d sample points along the scan line. The length of the
+        profile is the ceil of the computed length of the scan line.
```

Comment:
If I understand correctly, N is the length of the profile, and M is the number of points sampled in each "disk" of the cylinder? Or vice versa... State this in the docstring, both for inputs and outputs.